src/uwtools/cli.py

```diff
@@ -305,6 +306,89 @@ def _dispatch_forecast_run(args: Namespace) -> bool:
     )
 
 
+# Mode rocoto
+
+
+def _add_subparser_rocoto(subparsers: Subparsers) -> ModeChecks:
+    """
+    Subparser for mode: rocoto
+
+    :param subparsers: Parent parser's subparsers, to add this subparser to.
+    """
+    parser = _add_subparser(subparsers, STR.rocoto, "Realize and validate Rocoto XML Documents")
+    _basic_setup(parser)
+    subparsers = _add_subparsers(parser, STR.submode)
+    return {
+        STR.realize: _add_subparser_rocoto_realize(subparsers),
+        STR.validate: _add_subparser_rocoto_validate(subparsers),
+    }
+
+
+def _add_subparser_rocoto_realize(subparsers: Subparsers) -> SubmodeChecks:
+    """
+    Subparser for mode: rocoto realize
+
+    :param subparsers: Parent parser's subparsers, to add this subparser to.
+    """
+    parser = _add_subparser(subparsers, STR.realize, "Realize a Rocoto XML workflow document")
+    required = parser.add_argument_group(TITLE_REQ_ARG)
+    _add_arg_output_file(required)
```

Comment:
The flip side of read-input-file vs read-from-stdin option would be a write-output-file vs write-to-stdout option. Ideally we'd want to be able to e.g.
```
cat some.yaml | uw rocoto realize >some.xml
```
making use of the stdin/stdout streams. That's probably beyond the scope of this PR, but would you mind adding a backlog task for it?